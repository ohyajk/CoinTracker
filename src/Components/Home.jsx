import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { addCoin } from '../Redux/Coins/CoinSlice';

const Home = () => {
  const [query, setQuery] = useState('');
  const dispatch = useDispatch();

  const isLoading = useSelector((state) => state.coins.loading);
  const allCoins = useSelector((state) => state.coins.data.coins);

  return (
    <div data-testid="home-1" className="">
      <div className="w-[100%] flex justify-center items-center">
        <form className="text-white flex items-center justify-center w-[90%] md:w-[60%] m-2">
          <input onKeyUp={(e) => { setQuery(e.target.value.toLowerCase()); }} className="bg-[#c74076] px-4 py-2 outline-none w-[80%] border border-white rounded-l-full" type="text" placeholder="Search Coin name" required />
          <input onClick={(e) => { e.preventDefault(); setQuery(query); }} className="py-2 px-4 bg-[#dd4984] border border-white hover:bg-[#ce457b] sm:w-[20%] cursor-pointer rounded-r-full" type="button" value="Search" />
        </form>
      </div>
      <div>
        {isLoading ? <span className="text-center text-2xl flex items-center justify-center h-[40vh]">Loading ! Please Wait...</span> : ''}
        <ul className="grid grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 clr [&>*:nth-child(4n+2)]:bg-[#d1447a] [&>*:nth-child(4n-1)]:bg-[#d1447a]">
          {allCoins?.filter((coin) => (coin.name.toLowerCase().includes(query)))
            .map((coinf) => (
              <Link to="/coin" onClick={() => { dispatch(addCoin(coinf)); }} key={coinf.id}>
                <li className="flex flex-col items-start py-2 px-4 m-1 ">
                  <i className="fa-solid fa-circle-arrow-right ml-auto" />
                  <img className="h-16" src={coinf.icon} alt={coinf.id} />
                  <h2 className="md:text-xl font-bold ml-auto">{coinf.name}</h2>
                  <span className={coinf.priceChange1d < 0 ? 'text-red-400 text-sm ml-auto' : 'text-green-400 text-sm ml-auto'}>
                    {coinf.priceChange1d}
                    {' '}
                    <i className={coinf.priceChange1d < 0 ? 'fa-solid fa-arrow-trend-down' : 'fa-solid fa-arrow-trend-up'} />
                  </span>
                </li>
              </Link>
            ))}
        </ul>
        {' '}
        {!allCoins && isLoading === false ? (
          <div className="text-center text-2xl flex items-center justify-center h-[40vh] ">
            <span>
              <span className="changer">Oh Snap !</span>
              {' '}
              We have a problem...
            </span>
          </div>
        ) : ''}
      </div>

    </div>
  );
};

export default Home;
