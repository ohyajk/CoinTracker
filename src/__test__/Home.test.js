import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/ConfigureStore';
import Home from '../Components/Home';
import '@testing-library/jest-dom/extend-expect';

test('should render Home', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

test('should render snap', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('home-1')).toHaveTextContent('Snap');
});
