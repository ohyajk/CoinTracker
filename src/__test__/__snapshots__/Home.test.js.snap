// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        data-testid="home-1"
      >
        <div
          class="w-[100%] flex justify-center items-center"
        >
          <form
            class="text-white flex items-center justify-center w-[90%] md:w-[60%] m-2"
          >
            <input
              class="bg-[#0d0b13] px-4 py-2 outline-none w-[80%] border border-white/50 rounded-l-full focus:border-violet-600/50 focus:shadow-sm shadow-violet-400"
              placeholder="Search Coin name"
              required=""
              type="text"
            />
            <input
              class="py-2 px-4 bg-[#1a1729] border border-white/50 hover:bg-violet-400/75 sm:w-[20%] cursor-pointer rounded-r-full"
              type="button"
              value="Search"
            />
          </form>
        </div>
        <div>
          <ul
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
          />
           
          <div
            class="text-center text-2xl flex items-center justify-center h-[40vh] "
          >
            <span>
              <span
                class="changer"
              >
                Oh Snap !
              </span>
               
              We have a problem...
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
      data-testid="home-1"
    >
      <div
        class="w-[100%] flex justify-center items-center"
      >
        <form
          class="text-white flex items-center justify-center w-[90%] md:w-[60%] m-2"
        >
          <input
            class="bg-[#0d0b13] px-4 py-2 outline-none w-[80%] border border-white/50 rounded-l-full focus:border-violet-600/50 focus:shadow-sm shadow-violet-400"
            placeholder="Search Coin name"
            required=""
            type="text"
          />
          <input
            class="py-2 px-4 bg-[#1a1729] border border-white/50 hover:bg-violet-400/75 sm:w-[20%] cursor-pointer rounded-r-full"
            type="button"
            value="Search"
          />
        </form>
      </div>
      <div>
        <ul
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
        />
         
        <div
          class="text-center text-2xl flex items-center justify-center h-[40vh] "
        >
          <span>
            <span
              class="changer"
            >
              Oh Snap !
            </span>
             
            We have a problem...
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
